tests/cpp/runtime/contrib/ethosn/inference_test.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file tests/cpp/runtime/contrib/ethosn/inference_test.cc
+ * \brief Tests to check runtime components used during inference.
+ */
+
+#ifdef ETHOSN_HW
+
+#include <gtest/gtest.h>
+
+#include "../../../../../src/runtime/contrib/ethosn/ethosn_device.cc"
+
+namespace tvm {
+namespace runtime {
+namespace ethosn {
+
+TEST(WaitForInference, FailedResultRead) {
+  const int inference_error = 3;
+  const int timeout = 0;
+  dl::Inference inference = dl::Inference(inference_error);
+  WaitStatus result = WaitForInference(&inference, timeout);
+
+  ASSERT_EQ(result.GetErrorCode(), WaitErrorCode::Error);
+  ICHECK_EQ(result.GetErrorDescription(),
+            "Failed to read inference result status (No such file or directory)");
+}
+
+TEST(WaitForInference, InferenceTimeout) {
+  const int inference_scheduled = 0;
+  const int timeout = 0;
+  dl::Inference inference = dl::Inference(inference_scheduled);
```

Comment:
just curious why we don't have tests for other inputs except 0 and 3?

Reply:
No particular reason other than being focussed on checking the error messages, will add those!

Reply:
Added - 2 would be *success* which is tricky to test at this point due to requiring a valid `Inference` object